flow/project.py

```diff
@@ -2694,12 +2865,58 @@ def _verify_group_compatibility(self, groups):
         """Verifies that all selected groups can be submitted together."""
         return all(a.isdisjoint(b) for a in groups for b in groups if a != b)
 
-    def _verify_aggregate_project(self, aggregate):
-        """Verifies that all aggregates belongs to the same project."""
-        for job in aggregate:
-            if job not in self:
-                raise ValueError("Job {} is not present "
-                                 "in the project".format(job))
+    def _aggregate_is_in_project(self, aggregate):
+        """Verifies that the aggregate belongs to the this project."""
+        return any(aggregate in aggregates for aggregates in self._stored_aggregates)
+
+    @staticmethod
+    def _is_selected_aggregate(aggregate, jobs):
+        """Verifies whether the aggregate is present in the provided jobs.
+
+        Providing ``jobs=None`` indicates that no specific job is provided by
+        the user and hence ``aggregate`` is eligible for further evaluation.
+
+        Always returns True if jobs is None.
+        """
+        return (jobs is None) or (aggregate in jobs)
+
+    def _get_aggregate_from_id(self, id):
+        # Iterate over all the instances of stored aggregates and search for the
+        # aggregate in those instances.
+        for aggregates in self._stored_aggregates:
+            try:
```

Comment:
Given that this is excepted to error somewhat frequently we will want to change this to an if statement as try blocks are more costly.

Reply:
I fixed this.